src/Device/jprogOperations.js

```diff
@@ -138,20 +140,27 @@ const verifySerialPortAvailable = device => {
  * @returns {Promise} Promise that resolves if successful or rejects with error.
  */
 async function validateFirmware(device, fwVersion) {
+    let valid = false;
     try {
         const fwInfo = await nrfDeviceLib.readFwInfo(
-            getDeviceLibContext(),
+            deviceLibContext,
             device.id
         );
-        const valid = fwInfo.imageInfoList.find(imageInfo =>
-            imageInfo.version.string.includes(fwVersion)
-        );
+        if (
+            typeof fwVersion === 'object' &&
+            typeof fwVersion.validator === 'function'
+        ) {
+            valid = fwVersion.validator(fwInfo.imageInfoList);
+        }
+        valid = fwInfo.imageInfoList.find(imageInfo => {
+            if (!imageInfo.version.string) return false;
+            return imageInfo.version.string.includes(fwVersion);
+        });
         await reset(device.id); // Remove after fixing in nrf-device-lib
-        if (valid) return true;
+        return valid;
```

Comment:
Could this simply be
```
return fwInfo.imageInfoList.find(imageInfo => {
            if (!imageInfo.version.string) return false;
            return imageInfo.version.string.includes(fwVersion);
        });
```

Reply:
Originally that would be the case, but then we discovered that the BLE app validates the firmware differently, i.e. `fwVersion` is an object with a `validator` function, not a string, so this comparison would always fail in the BLE app.